### ISSUE TITLE
Minor - JSONObject.put(String Object) throws wrong Exception

### ISSUE THREAD

**Issue by krakenreleased** [stleary/JSON-java#402]
The javadoc states, invoking JSONObject.put(String key, Object value) with a null key should throw a JSONException, however it is throwing a null pointer exception instead.

```**
     * Put a key/value pair in the JSONObject. If the value is null, then the
     * key will be removed from the JSONObject if it is present.
     *
     * @param key
     *            A key string.
     * @param value
     *            An object which is the value. It should be of one of these
     *            types: Boolean, Double, Integer, JSONArray, JSONObject, Long,
     *            String, or the JSONObject.NULL object.
     * @return this.
     * @throws JSONException
     *             If the value is non-finite number or if the key is null.
     */
    public JSONObject put(String key, Object value) throws JSONException {
        if (key == null) {
            throw new NullPointerException("Null key.");
        }
        if (value != null) {
            testValidity(value);
            this.map.put(key, value);
        } else {
            this.remove(key);
        }
        return this;
    }
```

**Comment by johnjaylward** [comment]
I will have a patch for this soon.

**Comment by johnjaylward** [comment]
Actually, @stleary, Douglas was the last one to touch both the JavaDoc and the code that throws the exception. He changed it way back in a73066f6bacd7c2fbb545dbbc2be3e569493b0b5 from throwing a JSONException to throwing the NPE.

I'd say we defer to the JavaDoc on this one, but perhaps a simple JavaDoc update is all that's required. Sadly the commit message only really mentions one of many changes that were applied (mostly formatting), so I'm not sure if the code change was intended or not,

**Comment by stleary** [comment]
@johnjaylward Thanks for looking up the history. No preference. Leaving the NullPointerException alone is safer, changing to JSONException maintains consistency with the most of the rest of the lib, other than JSONPointer.

**Comment by stleary** [comment]
I believe this has been addressed by #405. If you think otherwise, let me know.
